### DISCUSSION TITLE
kernelTestCase test class error

### DISCUSSION THREAD

**Discussion by Elmatbaisaad** [symfony/symfony#46456] [Q&A]
Hello everyone 
i am trying to use KernelTestCase to test database in my project but i keep getting this errors:
![image](https://user-images.githubusercontent.com/101296742/170217010-e948b9f4-da4c-41f2-8964-9c376e95845d.png)
here is my env:
![image](https://user-images.githubusercontent.com/101296742/170217119-97b1be76-d199-4293-b7fb-495430b072b5.png)
the DatabasePrimer Classe:
![image](https://user-images.githubusercontent.com/101296742/170217256-b6994e27-bfe9-46f2-8946-d500bb97de8a.png)

**Comment by stof** [comment]
Well, make sure that the `mysql` host resolves properly. The error you get is a DNS resolution error.

**Reply by Elmatbaisaad** [reply]
thank you for your answer how can i make sure of that please ?

**Comment by ghost** [comment]
Check whether your PHP container is in the same docker network as mysql container. However if you don't have the full containerization, i.e. PHP is installed on your computer and only mysql is running inside docker container then it will not work as `mysql` is a network alias for the container but inside docker network.

**Reply by Elmatbaisaad** [reply]
Hi @javaDeveloperKid thank you for your answer.
I solved the probleme by using mocks and stubs to do the test because when i test i wont touch the production database
